tests/unit/amo/reducers/test_addonsByAuthors.js

```diff
@@ -0,0 +1,129 @@
+  loadOtherAddonsByAuthors,
+} from 'amo/reducers/addonsByAuthors';
+import { fakeAddon } from 'tests/unit/amo/helpers';
+
+
+describe(__filename, () => {
+  describe('reducer', () => {
+    it('initializes properly', () => {
+      const state = reducer(undefined);
+      expect(state).toEqual(initialState);
+    });
+
+    it('ignore unrelated actions', () => {
+      const state = {
+        ...initialState,
+        // We slightly modify the state to be sure that an unrelated action
+        // does not change it.
+        byAddon: {
+          'addon-slug': {},
+        },
+      };
+      const newState = reducer(state, { type: 'UNRELATED' });
+      expect(newState).toEqual(state);
+    });
+
+    describe('LOAD_OTHER_ADDONS_BY_AUTHORS', () => {
+      it('allows an empty list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [],
+        }));
+        expect(state.byAddon).toEqual({
+          'addon-slug': [],
+        });
+      });
+
+      it('adds related add-ons by slug', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: 'addon-slug',
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddon).toEqual({
+          'addon-slug': [fakeAddon],
+        });
+      });
+
+      it('excludes the main add-on from the list of add-ons', () => {
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug: fakeAddon.slug,
+          addons: [fakeAddon],
+        }));
+        expect(state.byAddon).toEqual({
+          [fakeAddon.slug]: [],
+        });
+      });
+
+      it('adds OTHERS_ADDONS_BY_AUTHORS_TO_LOAD to the state', () => {
+        const slug = 'addon-slug';
+        const state = reducer(undefined, loadOtherAddonsByAuthors({
+          slug,
+          // This is the case where there are more add-ons loaded than needed.
+          addons: Array(OTHERS_ADDONS_BY_AUTHORS_TO_LOAD + 2).fill(fakeAddon),
+        }));
+        expect(state.byAddon[slug])
+          .toHaveLength(OTHERS_ADDONS_BY_AUTHORS_TO_LOAD);
+      });
+    });
+  });
+
+  describe('fetchOtherAddonsByAuthors()', () => {
+    it('throws an error if arguments are missing', () => {
+      expect(() => {
+        fetchOtherAddonsByAuthors({});
+      }).toThrow(/errorHandlerId is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({ errorHandlerId: 'id' });
+      }).toThrow(/An add-on slug is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({ errorHandlerId: 'id', slug: 'addon-slug' });
+      }).toThrow(/An add-on type is required/);
+      expect(() => {
+        fetchOtherAddonsByAuthors({
+          errorHandlerId: 'id',
+          slug: 'addon-slug',
+          type: ADDON_TYPE_THEME,
+          addonType: ADDON_TYPE_THEME,
+        });
+      }).toThrow(/Authors are required/);
+    });
+
+    it('returns an array of authors in payload', () => {
```

Comment:
I don't think this kind of test is very useful. You will be covering this logic in the saga test.